### PR TITLE
Integration front-end and back-end

### PR DESCRIPTION
**Pull request by PsyhoBelka**
- create back-end nodejs express server
- config routes for general collection (animals\news\supplies\happy stories\about\donate)
- config&connect external MongoDB database
- create initialization scripts for prepopulate data into DB
- replace json-mock in front-end to back-end endpoints

### CODE REVIEW

**Comment by maksv21** [comment]
How to start the backend server? It does not start automatically, so the data cannot be downloaded
image
image

At SS-ITA-JS_RightForTheLife/right-for-life_front/src/containers/HomePage/HomePage.jsx 24:1 replace 
`const dataApi = 'https://raw.githubusercontent.com/AlexeyKasaev3/softServe-academy/master/demo-3-data/news.json';
`

**Comment by yevhverb** [comment]
> 1. How to start the backend server? It does not start automatically, so the data cannot be downloaded
>    image
>    image
> 2. At SS-ITA-JS_RightForTheLife/right-for-life_front/src/containers/HomePage/HomePage.jsx 24:1 replace
>    `const dataApi = 'https://raw.githubusercontent.com/AlexeyKasaev3/softServe-academy/master/demo-3-data/news.json'; `

You can open the “right-for-life_back” folder and run the “**npm start**” script, but before of them you should install nodemon package "**npm install nodemon**".

**Comment by maksv21** [comment]
> you should install nodemon package - "**npm install nodemon**".

Thanks, only nodemon was missing to start

So, only needed:

add 
> nodemon in the dev dependencies field

and 


> At SS-ITA-JS_RightForTheLife/right-for-life_front/src/containers/HomePage/HomePage.jsx 24:1 replace
>     `const dataApi = 'https://raw.githubusercontent.com/AlexeyKasaev3/softServe-academy/master/demo-3-data/news.json';`

**Comment by PsyhoBelka** [comment]
@yevhverb thanks for comment!
nodemon was added to dependencies

**Comment by PsyhoBelka** [comment]
> probably notifications doesn't come from simple comments, so I'll duplicate it
> edit At SS-ITA-JS_RightForTheLife/right-for-life_front/src/containers/HomePage/HomePage.jsx 24:17

make no sense due to mock with Load indicator

File: right-for-life_back/app.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const express = require('express');
+const morgan = require('morgan');
+const bodyParser = require('body-parser');
+const animalRoutes = require('./routes/animalsRoutes.js');
+const newsRoutes = require('./routes/newsRoutes.js');
+const happyStoriesRoutes = require('./routes/happyStoriesRoutes.js');
+const suppliesRoutes = require('./routes/supliesRoutes.js');
+const aboutRoutes = require('./routes/aboutRoutes.js');
+const donateRoutes = require('./routes/donatesRoutes.js');
```
Comment (protonaby) [inline right-for-life_back/app.js]: Could you please also add Model and Route for EmergencyHelpPage and SingleEmergencyHelpPage?

File: right-for-life_back/utils/configs.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const dbName = 'right-for-life';
+const dbOptions = {
+  useUnifiedTopology: true,
+  useNewUrlParser: true,
+  dbName: dbName,
+};
+
+module.exports.dbName = dbName;
+module.exports.dbURL = `mongodb://localhost:27017/${dbName}`;
```
Comment (protonaby) [inline right-for-life_back/utils/configs.js]: Sorry if dumb question, but if we use external Mongo DB, why it's localhost here?

File: right-for-life_front/src/app/App.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      animals: [],
+      textId: '',
+    };
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:4000/animals')
```
Comment (protonaby) [inline right-for-life_front/src/app/App.js]: You may use BE_URL constant for http://localhost:4000 part

File: right-for-life_back/utils/configs.js (hunk 2 above)
Comment (PsyhoBelka) [reply right-for-life_back/utils/configs.js]: > Sorry if dumb question, but if we use external Mongo DB, why it's localhost here?

it didn't used right now:)

File: right-for-life_front/src/app/App.js (hunk 3 above)
Comment (PsyhoBelka) [reply right-for-life_front/src/app/App.js]: missed! thnx!

File: right-for-life_back/app.js (hunk 1 above)
Comment (PsyhoBelka) [reply right-for-life_back/app.js]: could you pls provide json-example (or maybe some initial data) for them?

Comment (protonaby) [reply right-for-life_back/app.js]: JSON can be found at https://alex-boklag.github.io/SSA-Demo-AnimalShelter/db/emergency.json

Comment (PsyhoBelka) [reply right-for-life_back/app.js]: added

**Pull request merged**